[Alvozyn] iP

Alvozyn opened the pull request:
## Duke 
>Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke helps you remember all the task that you need to do :+1:
- text-based
- easy to learn
- ~FAST~ __SUPER__ *FAST*

All you need to do is:
1. Download [here](https://github.com/Alvozyn/ip/releases/tag/v0.2)!
2. Place it in a folder
3. Open your terminal and navigate to the folder
4. Enter in  ```java -jar duke.jar```

Features:
- [x] Add task
- [x] Delete task
- [x] Mark task
- [x] Unmark task
- [x] Find task

Here is the ```main``` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Level-1:Greet, Echo, Exit

Commit:
Level-2: Add, List

Commit:
Level-3: Mark as Done
Added a new Task class and modified toDoList to store the new Task class

Alvozyn committed:
Duke Level-4: ToDo, Event, Deadline

Alvozyn committed:
Level-4: ToDo, Event, Deadline
Make the classes to be inheritance instead

Alvozyn committed:
Sample Automated Text UI Testing

Alvozyn committed:
Level-5: Handle Errors

Alvozyn committed:
Level-5: Handling Errors

Alvozyn committed:
Level-6: Delete

Alvozyn committed:
Refractor code
Move all the codes under one try catch block instead of having multiple of them.

Alvozyn committed:
Add Level-7: Save
Users can now save and load a task list.

Alvozyn committed:
Add headers
Add headers for the updateFile and loadFile

Alvozyn committed:
Add Level-8
Timings are now stored as Date instead of String

Alvozyn committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add Level-8 Timings are now stored as Date instead of String

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/toDoList.java

Alvozyn committed:
Add classes Ui, Parser, DateTime, Storage and TaskList

Alvozyn committed:
Divide classes into packages

Alvozyn committed:
Remove headers

Alvozyn committed:
Merge branch 'branch-Level-7'

Alvozyn committed:
Remove redundant class

Alvozyn committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Alvozyn committed:
Add Command, AddCommand, DeleteCommand, ExitCommand, InvalidCommand, ListCommand, MarkCommand and UnmarkCommand classes and sort them into packages

Alvozyn committed:
Remove redundant variable and constructer

Alvozyn committed:
Fix some bugs in the Duke class

Alvozyn committed:
Add JUnit Tests

YuanXun9834 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+
+    public Ui ui;
+    public Storage storage;
+    public TaskList listOfTasks;
```
I think a better way to name the TaskList variable would be just tasks

YuanXun9834 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,170 @@
+        this.ui = ui;
+    }
+
+    public Command commandExecute(String input, TaskList toDo, Storage storage) {
```
Maybe the method name here should be executeCommand as suggested by our coding standard to make it sound like a verb

YuanXun9834 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,170 @@
+
+                return new MarkCommand(index);
+
+            } else if (instruction[0].equals("unmark")) { //unmarking
```
Since you already have a command class, maybe abstracting out different commands as different classes to make use of dynamic binding to execute commands.

YuanXun9834 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+        allTasks =  new ArrayList<Task>();
+    }
+
+    public int numberOfTask() {
```
method names should start with a verb! 👍

YuanXun9834 reviewed (commented):
Overall great attempt at trying to follow the coding standards, hope some of my suggestions were useful :)

Alvozyn committed:
Add javadocs

Alvozyn committed:
Refractor code and remove unnecessary class

Alvozyn committed:
Add Level-9

Alvozyn committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Refractor code and remove unnecessary class

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/task/Task.java

Alvozyn committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Level-9

# Conflicts:
#	src/main/java/duke/TaskList.java

Alvozyn committed:
Add javadocs

Alvozyn committed:
Add additional plugins and change the mainClassName

GlennOngJunJie commented on src/main/java/duke/Parser.java:
```diff
@@ -1,4 +1,4 @@
-
+package duke;
```
I understand that you might only have one package, but can consider importing a specific class in the package like, duke.Tasklist and duke.Storage

GlennOngJunJie commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,164 @@
+
+                System.out.println("Now you have " + toDo.numberOfTask() + " tasks in the list.");
+
+            }
```
Good job here! Another alternative might be to consider using switch to make the code more readable, but overall still good job :)

GlennOngJunJie commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,31 @@
+package duke.command;
```
Really love the use of a separate package for command! Well done :)

GlennOngJunJie reviewed (commented):
Overall very very good job, and I have definitely learn some things from you that can help make my code clearer and better! :D

Alvozyn committed:
Add Gui

Alvozyn committed:
Fix bug at ParserTest

Alvozyn committed:
Update GUI

Alvozyn committed:
Update GUI layout

Alvozyn committed:
Shorten the parse method

The parse method is too long.

The method is short by creating sub methods to check the validity of the user input instead.

Sub methods allow the user to read the code much more easily

The method created are checkMark, checkUnmark, checkDelete, checkFind, checkBasicAddTask, checkDeadline and checkEvent.

Alvozyn committed:
Remove if else statements

Switch is faster than if else.

Updated the if else statement to switch statements.

The bot recognise many command so switch statements are faster as the compiler creates a jump table for a switch during compilation.

Alvozyn committed:
Add Assertion

Alvozyn committed:
Merge pull request #2 from Alvozyn/branch-A-Assertions

Add Assertion

Alvozyn committed:
Remove dead codes, parameter and methods that are not in use and update javadoc

Alvozyn committed:
Merge pull request #3 from Alvozyn/branch-A-CodeQuality

Remove dead codes, irrelevant parameter and methods and update javadocs

Alvozyn committed:
Implement AddNote command
User can now add notes under the task

Alvozyn committed:
Add ListNoteCommand for D-Notes

Alvozyn committed:
Add DeleteNoteCommand for D-Notes

Alvozyn committed:
Merge remote-tracking branch 'origin/branch-BCD-Extension'

* origin/branch-BCD-Extension:
  Add DeleteNoteCommand for D-Notes
  Add ListNoteCommand for D-Notes
  Add AddNote command for D-Notes

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java

Alvozyn committed:
Implement Help command for C-Help

Alvozyn committed:
Fix bugs

Alvozyn committed:
Update minor print out statements

Alvozyn committed:
Update GUI

Alvozyn committed:
Update README User Guide

Alvozyn committed:
Remove unnecessary code

Alvozyn committed:
Creates a file and the path does not exist

Alvozyn committed:
Update javadocs
